pkg/queue/sharedmain/main.go

```diff
@@ -110,22 +104,45 @@ type config struct {
 	// Concurrency State Endpoint configuration
 	ConcurrencyStateEndpoint  string `split_words:"true"` // optional
 	ConcurrencyStateTokenPath string `split_words:"true"` // optional
+
+	Env
+}
+
+type Env struct {
+	ServingNamespace     string `split_words:"true" required:"true"`
+	ServingRevision      string `split_words:"true" required:"true"`
+	ServingConfiguration string `split_words:"true" required:"true"`
+	ServingPodIP         string `split_words:"true" required:"true"`
+	ServingPod           string `split_words:"true" required:"true"`
+	ServingService       string `split_words:"true"` // optional
 }
 
+type Defaults struct {
+	Ctx       context.Context
+	Logger    *zap.SugaredLogger
+	Transport http.RoundTripper
+	Env       Env
+}
```

Comment:
This list includes two parts:
 - parameters whose value is determined by the service and provided for the extensions to use (but not to change)
 - parameters whose default is determined by the service and provided for the extensions such that may change them
 Maybe we should have divided this struct into two:
```
 type Defaults {
      Ctx              context.Context
      Transport    http.RoundTripper
 }	
 type Config {
     Env
     Logger  *zap.SugaredLogger
 }
 type QueueProxy struct {
        Invariables     Config
        Mutables        Defaults
 }
```

Reply:
I wouldn't split, I'd just note in comments which ones are considered mutable and which ones aren't. 

Since they're exported types, an overview comment on the `Defaults` and `Env` structs themselves wouldn't be a bad idea.